tt_eager/tt_dnn/op_library/bmm/multi_core_reuse_mcast_1d_optimized/bmm_op_multi_core_reuse_mcast_1d_optimized.cpp

```diff
@@ -1400,7 +1446,11 @@ operation::ProgramWithCallbacks create_program_mcast_in1(
             }
 
             if (src0_sharded) {
-                UpdateDynamicCircularBufferAddress(program, cb_src0, *src_buffer_a);
+                if (extract_shard_sub_blocks) {
+                    UpdateDynamicCircularBufferAddress(program, cb_src2, *src_buffer_a);
```

Comment:
Why is cb_src0 only updated when `extract_shard_sub_blocks` is false?

Isn't cb_src0 still used in the kernel you edited tt_eager/tt_dnn/op_library/bmm/kernels/dataflow/reader_bmm_tile_layout_in0_sender_padding.cpp around line 99?
```
    #ifdef IN0_SHARDED
    uint32_t in0_start_address = get_write_ptr(cb_id_in0);
    #endif
```

Reply:
It's also helpful to read the create program code, since the callback just follows whatever is done there.
<img width="1178" alt="image" src="https://github.com/tenstorrent/tt-metal/assets/109366241/3684e868-a391-4a4d-a084-ed41d89b00a5">

Reply:
Ok. Thanks.